### PR TITLE
feat: theming support with schematics

### LINKED ISSUE #2: Register tokens for npm, gh and others
Need 
`NPM_TOKEN` for publishing the library
`GH_TOKEN` for releases on gh side
`GH_NAME` and `GH_EMAIL`
`NX_CLOUD_AUTH_TOKEN` for Nx cloud

### PR DESCRIPTION
**Pull request by N1XUS**
## Related Issue(s)

 relates #9

## Description
* Added installation schematics which will add needed dependencies based on user answers;
* Added theming package to control multiple theming providers (ui5, fundamental, etc.);

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +37,7 @@
     "fast-deep-equal": "3.1.3",
     "fast-glob": "3.2.12",
     "lodash": "4.17.21",
+    "n1xus-ui5-theming-ngx": "latest",
```
Comment (N1XUS) [inline package.json:40]: This one is added by the schematics for local test app

File: package.json [hunk 2]
```diff
@@ -69,14 +71,15 @@
     "eslint": "~8.15.0",
     "eslint-config-prettier": "8.1.0",
     "eslint-plugin-cypress": "2.12.1",
+    "fundamental-styles": "^0.26.5",
```
Comment (N1XUS) [inline package.json:74]: We need this one to get the same array of themes supported by both libs - ui5 and fundamental-styles.

File: package.json [hunk 3]
```diff
@@ -24,6 +24,7 @@
     "@angular/platform-browser": "15.0.4",
     "@angular/platform-browser-dynamic": "15.0.4",
     "@angular/router": "15.0.4",
+    "@fundamental-styles/common-css": "^0.26.5",
```
Comment (N1XUS) [inline package.json:27]: This one is added by the schematics for local test app

File: libs/angular-generator/src/lib/theming/theming-models.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  getThemeNames(): string[] {
+    const nodeFs = new NodeFsImplementation();
+
+    const fdPath = nodeFs.normalize('node_modules/fundamental-styles/dist/theming/');
+
+    return nodeFs.queryFiles(`${fdPath}**.css`, []).map((themePath) => nodeFs.basename(themePath, nodeFs.extname(themePath)));
+  }
+
+  override getCode(): string {
+    this.addImport(['InjectionToken'], '@angular/core');
```
Comment (g-cheishvili) [inline libs/angular-generator/src/lib/theming/theming-models.ts:25]: I moved this addImports into other functions, after constructor step, getCode might be called several times and it is just redundant operations

File: libs/angular-generator/src/lib/theming/theming-module.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      local: 'Ui5ThemingModule',
+      exported: 'Ui5ThemingModule',
+      types: [ExportSpecifierType.Class],
+    });
+  }
+
+  override getCode(): string {
+    this.addImport(['NgModule', 'ModuleWithProviders', 'Optional', 'isDevMode', 'InjectionToken'], '@angular/core');
+    this.addImport(['Ui5ThemingService'], './ui5-theming.service');
+    this.addImport(['ThemingConfig', 'UI5_THEMING_CONFIGURATION'], './ui5-theming.models');
```
Comment (g-cheishvili) [inline libs/angular-generator/src/lib/theming/theming-module.ts:18]: Same thing about `addImport`

File: libs/angular-generator/src/lib/theming/theming-service.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    this.addExport({
+      local: 'Ui5ThemingService',
+      exported: 'Ui5ThemingService',
+      types: [ExportSpecifierType.Class],
+    });
+  }
+
+  override getCode(): string {
+    this.addImport(['Inject', 'Injectable', 'isDevMode'], '@angular/core');
+    this.addImport(['AvailableThemes', 'ThemingConfig', 'UI5_THEMING_CONFIGURATION', 'Ui5ThemingProvider', 'Ui5ThemingConsumer'], './ui5-theming.models');
```
Comment (g-cheishvili) [inline libs/angular-generator/src/lib/theming/theming-service.ts:17]: I would make second parameter more dynamic. If user renames `ui5-theming.models` file name, this code will not be able to detect that on generation

File: libs/angular-generator/src/lib/webcomponents-theming.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      local: 'Ui5WebcomponentsThemingService',
+      exported: 'Ui5WebcomponentsThemingService',
+      path: 'main/ui5-webcomponents-theming.service.ts',
+      types: [ExportSpecifierType.Class]
+    });
+  }
+
+  override getCode(): string {
+    this.addImport(['Injectable', 'OnDestroy', 'Optional'], '@angular/core');
+    this.addImport(['Ui5ThemingProvider', 'Ui5ThemingService', 'AvailableThemes'], '@ui5/theming-ngx');
```
Comment (g-cheishvili) [inline libs/angular-generator/src/lib/webcomponents-theming.ts:72]: Do we have theming-ngx package as well?

**Review by g-cheishvili — CHANGES_REQUESTED** [review]
I like it, just few minor things

File: libs/angular-generator/src/lib/webcomponents-theming.ts (hunk 7 above)
Comment (N1XUS) [reply libs/angular-generator/src/lib/webcomponents-theming.ts:72]: Yes. Its a global package that actually controls theming across multiple libraries that are not connected with each other

File: libs/angular-generator/src/lib/theming/theming-models.ts (hunk 4 above)
Comment (N1XUS) [reply libs/angular-generator/src/lib/theming/theming-models.ts:25]: Can't imagine a case when getCode would be called multiple times for one class instance. Could you please elaborate?

**Review by g-cheishvili — APPROVED** [review]

**Pull request merged**